packages/material-ui/src/Button/Button.js

```diff
@@ -162,6 +162,12 @@ export const styles = theme => ({
   focusVisible: {},
   /* Pseudo-class applied to the root element if `disabled={true}`. */
   disabled: {},
+  /* Styles applied to the root element if `loading={true}`. */
+  loading: {
```

Comment:
It's not actually related to loading (it's only used in this context). What this actually conveys is that the button is "deactivated" or "not actionable" or "temporary disabled". `loading` implies that there is another indicator.

As I said below I'd be perfectly fine with introducing a separate component that overrides `Button` properly.

Reply:
> As I said below I'd be perfectly fine with introducing a separate component that overrides Button properly.

The loading state is transient, how would you switch between the loading and non-loading state with a new component? In this regard, I think that a switch of a prop is simpler.

Reply:
I don't think `transient` applies to a loading state or any UI in that matter.

But even then you can still have the `loading` prop which then actually conveys the proper meaning by switching styles **and** adding an indicator.